Fix links and header case in key-event-migration guide (#9922)
flutter/website: src/release/breaking-changes/key-event-migration.md

framework and in client apps caused overly complex code, and the old system
didn't properly represent the true state of key events on the system.

> [`RawKeyboard`][] is the legacy API, and has been deprecated and
> will be removed in the future.
> The replacements are [`HardwareKeyboard`][] and [`KeyEvent`][] APIs
> (such as [`FocusNode.onKeyEvent`][]).

Reviewer: **chore:** Write in active voice per [GSG](https://developers.google.com/style/voice).
```suggestion
The legacy API [`RawKeyboard`][] has been deprecated and will be removed in the future.
The [`HardwareKeyboard`][] and [`KeyEvent`][] APIs replace this legacy API.
An example would be [`FocusNode.onKeyEvent`][].
```

> Behavior-wise, [`RawKeyboard`][] provided a
> less unified,less regular event model than [`HardwareKeyboard`][] does.
> For example:

Reviewer: **chore:** Don't make up words.
```suggestion
The behavior of [`RawKeyboard`][] provided a less unified and less regular event model
than [`HardwareKeyboard`][] does. Consider the following examples.
```

## Migration guide

> APIs provided by the Flutter framework are already migrated.
> Migration is needed only if any of the classes or methods listed in
> the previous section are used.

Reviewer: **chore:** Write in active voice per [GSG](https://developers.google.com/style/voice).
```suggestion
Flutter framework API have been migrated.
If your code includes any of the classes or methods listed in
the previous section, migrate to these new APIs.
```

}
```

> Be careful to check conditionals:
> a [`KeyRepeatEvent`][] is also a key down event, but
> it is a different type (it is not a subclass of [`KeyDownEvent`][]), so
> don't assume that `keyEvent is! KeyDownEvent` only allows key up events,
> because both `KeyDownEvent` and `KeyRepeatEvent` need to be checked.

Reviewer: **issue:** Write as shorter sentences per [GSG](https://developers.google.com/style/accessibility#ease-of-reading).

```suggestion
Be careful to check conditionals. 
Though not inheriting from [`KeyDownEvent`][],
a [`KeyRepeatEvent`][] event works like a key down event.
Don't assume that `keyEvent is! KeyDownEvent` only allows key up events.
Check both `KeyDownEvent` and `KeyRepeatEvent` events.
```
